### ISSUE TITLE
[source-bigquery] Fatal error when configuring connector in version 0.4.2 in Airbyte 0.50.38

### ISSUE THREAD

**Issue by icychocolate98** [airbytehq/airbyte#39422]
### Connector Name

source-bigquery

### Connector Version

0.4.2

### What step the error happened?

Configuring a new connector

### Relevant information

I'm having problems when configuring the BigQuery source on my different servers (in my local machine it's working OK)

The problem remains in the configuration step, where I'm receiving a fatal error on 'check' process, I think the error lies on the server CPU or something like that.

`platform > Fatal glibc error: CPU does not support x86-64-v2`

I suppose Airbyte version does not affect in this case, in my local machine I'm working with `0.58.1` and in production servers we are working with `0.50.43`

I'm pasting log output below.

### Relevant log output

```shell
2024-06-12 08:41:53 [46mplatform[0m > Docker volume job log path: /tmp/workspace/36a2f293-c083-494a-83ab-270ef99d7781/0/logs.log
2024-06-12 08:41:53 [46mplatform[0m > Executing worker wrapper. Airbyte version: 0.50.38
2024-06-12 08:41:53 [46mplatform[0m > Attempt 0 to save workflow id for cancellation
2024-06-12 08:41:53 [46mplatform[0m > Using default value for environment variable SIDECAR_KUBE_CPU_LIMIT: '2.0'
2024-06-12 08:41:53 [46mplatform[0m > 
2024-06-12 08:41:53 [46mplatform[0m > ----- START CHECK -----
2024-06-12 08:41:53 [46mplatform[0m > Using default value for environment variable SOCAT_KUBE_CPU_LIMIT: '2.0'
2024-06-12 08:41:53 [46mplatform[0m > 
2024-06-12 08:41:53 [46mplatform[0m > Using default value for environment variable SIDECAR_KUBE_CPU_REQUEST: '0.1'
2024-06-12 08:41:53 [46mplatform[0m > Using default value for environment variable SOCAT_KUBE_CPU_REQUEST: '0.1'
2024-06-12 08:41:53 [46mplatform[0m > Using default value for environment variable LAUNCHDARKLY_KEY: ''
2024-06-12 08:41:53 [46mplatform[0m > Checking if airbyte/source-bigquery:0.4.2 exists...
2024-06-12 08:41:53 [46mplatform[0m > airbyte/source-bigquery:0.4.2 was found locally.
2024-06-12 08:41:53 [46mplatform[0m > Creating docker container = source-bigquery-check-36a2f293-c083-494a-83ab-270ef99d7781-0-rkzvb with resources io.airbyte.config.ResourceRequirements@4cba9720[cpuRequest=,cpuLimit=,memoryRequest=,memoryLimit=,additionalProperties={}] and allowedHosts null
2024-06-12 08:41:53 [46mplatform[0m > Preparing command: docker run --rm --init -i -w /data/36a2f293-c083-494a-83ab-270ef99d7781/0 --log-driver none --name source-bigquery-check-36a2f293-c083-494a-83ab-270ef99d7781-0-rkzvb --network host -v airbyte_workspace:/data -v /tmp/airbyte_local:/local -e DEPLOYMENT_MODE=OSS -e WORKER_CONNECTOR_IMAGE=airbyte/source-bigquery:0.4.2 -e AUTO_DETECT_SCHEMA=true -e LAUNCHDARKLY_KEY= -e SOCAT_KUBE_CPU_REQUEST=0.1 -e SOCAT_KUBE_CPU_LIMIT=2.0 -e FIELD_SELECTION_WORKSPACES= -e USE_STREAM_CAPABLE_STATE=true -e AIRBYTE_ROLE= -e WORKER_ENVIRONMENT=DOCKER -e APPLY_FIELD_SELECTION=false -e WORKER_JOB_ATTEMPT=0 -e OTEL_COLLECTOR_ENDPOINT=http://host.docker.internal:4317 -e FEATURE_FLAG_CLIENT=config -e AIRBYTE_VERSION=0.50.38 -e WORKER_JOB_ID=36a2f293-c083-494a-83ab-270ef99d7781 airbyte/source-bigquery:0.4.2 check --config source_config.json
2024-06-12 08:41:53 [46mplatform[0m > Reading messages from protocol version 0.2.0
2024-06-12 08:41:53 [46mplatform[0m > Fatal glibc error: CPU does not support x86-64-v2
2024-06-12 08:41:53 [46mplatform[0m > Check connection job subprocess finished with exit code 127
2024-06-12 08:41:53 [46mplatform[0m > Unexpected error while checking connection: 
io.airbyte.workers.exception.WorkerException: Error checking connection status: no status nor failure reason were outputted
	at io.airbyte.workers.WorkerUtils.throwWorkerException(WorkerUtils.java:268) ~[io.airbyte-airbyte-commons-worker-0.50.38.jar:?]
	at io.airbyte.workers.general.DefaultCheckConnectionWorker.run(DefaultCheckConnectionWorker.java:120) ~[io.airbyte-airbyte-commons-worker-0.50.38.jar:?]
	at io.airbyte.workers.general.DefaultCheckConnectionWorker.run(DefaultCheckConnectionWorker.java:44) ~[io.airbyte-airbyte-commons-worker-0.50.38.jar:?]
	at io.airbyte.workers.temporal.TemporalAttemptExecution.get(TemporalAttemptExecution.java:142) ~[io.airbyte-airbyte-workers-0.50.38.jar:?]
	at io.airbyte.workers.temporal.check.connection.CheckConnectionActivityImpl.lambda$runWithJobOutput$1(CheckConnectionActivityImpl.java:138) ~[io.airbyte-airbyte-workers-0.50.38.jar:?]
	at io.airbyte.commons.temporal.HeartbeatUtils.withBackgroundHeartbeat(HeartbeatUtils.java:57) ~[io.airbyte-airbyte-commons-temporal-core-0.50.38.jar:?]
	at io.airbyte.workers.temporal.check.connection.CheckConnectionActivityImpl.runWithJobOutput(CheckConnectionActivityImpl.java:123) ~[io.airbyte-airbyte-workers-0.50.38.jar:?]
	at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:580) ~[?:?]
	at io.temporal.internal.activity.RootActivityInboundCallsInterceptor$POJOActivityInboundCallsInterceptor.executeActivity(RootActivityInboundCallsInterceptor.java:64) ~[temporal-sdk-1.17.0.jar:?]
	at io.temporal.internal.activity.RootActivityInboundCallsInterceptor.execute(RootActivityInboundCallsInterceptor.java:43) ~[temporal-sdk-1.17.0.jar:?]
	at io.temporal.internal.activity.ActivityTaskExecutors$BaseActivityTaskExecutor.execute(ActivityTaskExecutors.java:95) ~[temporal-sdk-1.17.0.jar:?]
	at io.temporal.internal.activity.ActivityTaskHandlerImpl.handle(ActivityTaskHandlerImpl.java:92) ~[temporal-sdk-1.17.0.jar:?]
	at io.temporal.internal.worker.ActivityWorker$TaskHandlerImpl.handleActivity(ActivityWorker.java:241) ~[temporal-sdk-1.17.0.jar:?]
	at io.temporal.internal.worker.ActivityWorker$TaskHandlerImpl.handle(ActivityWorker.java:206) ~[temporal-sdk-1.17.0.jar:?]
	at io.temporal.internal.worker.ActivityWorker$TaskHandlerImpl.handle(ActivityWorker.java:179) ~[temporal-sdk-1.17.0.jar:?]
	at io.temporal.internal.worker.PollTaskExecutor.lambda$process$0(PollTaskExecutor.java:93) ~[temporal-sdk-1.17.0.jar:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642) ~[?:?]
	at java.lang.Thread.run(Thread.java:1583) ~[?:?]
2024-06-12 08:41:53 [46mplatform[0m > 
2024-06-12 08:41:53 [46mplatform[0m > ----- END CHECK -----
2024-06-12 08:41:53 [46mplatform[0m >
```

### Contribute

- [ ] Yes, I want to contribute

**Comment by icychocolate98** [comment]
Good afternon @marcosmarxm ! 

We are running a Debian server (a virtual machine)

`Linux "server-hostname" 5.10.0-29-amd64 #1 SMP Debian 5.10.216-1 (2024-05-03) x86_64 GNU/Linux`


Thank you in advice !

**Comment by icychocolate98** [comment]
Hi everyone,

After reviewing and conducting some tests, we identified that the issue was related to the virtual machine configuration where the server is running.

By default, the server was set to use the 'kvm64' CPU emulation. However, changing this setting to 'host' resolved the issue.

So, we accessed the server, changed virtual machine configuration and that's all. 

Since then, the issue has not reoccurred.

We hope this helps others who might be facing the same problem.
